pennylane/devices/default_qubit.py

```diff
@@ -202,6 +202,9 @@ def adjoint_state_measurements(
 
 def adjoint_ops(op: qml.operation.Operator) -> bool:
     """Specify whether or not an Operator is supported by adjoint differentiation."""
+    if isinstance(op, qml.ops.Exp) and qml.math.is_abstract(op.data[0]):
+        # Skip validation of Exp in tracing because Exp.has_generator is not traceable.
+        return True
```

Comment:
Should I return `True` by default or `False`? What makes more sense here?

Reply:
Is `Exp` natively supported in lightning? If not I would think `False` would be correct here. Even it is supported I don't know if lightning has a generator defined for it 🤔

Reply:
This is defined in `default_qubit.py`, not sure if we should base its logic on what Lightning supports? Is this function used anywhere else other than in Lightning?

Reply:
I'm leaning more towards `True`, assuming that an error gets raised at some point when the coefficient is invalid with jitting. The reason is that I don't think we should unconditionally restrict users from using `Exp` in a circuit when jitting.

However, if an error is not raised at any point with jit during execution, then I think we should go with `False`. This would trigger decomposition.

Reply:
> I'm leaning more towards True, assuming that an error gets raised at some point when the coefficient is invalid with jitting. The reason is that I don't think we should unconditionally restrict users from using Exp in a circuit when jitting.

I agree with what you are saying, but this function should determine whether this operator is adjoint differentiable, not determine whether or not it can be used in a circuit when jitting.

Reply:
The existing logic returns `True` if `op.num_params == 1 and op.has_generator`, where `op.has_generator` for `Exp` returns `self.base.is_hermitian and not np.real(self.coeff)`. What I could do is, instead of returning `True` or `False`, I return `op.num_params == 1 and op.base.is_hermitian`

Reply:
I moved the check to `Exp.has_generator`